src/rgw/rgw_op.cc

```diff
@@ -2262,13 +2284,12 @@ void RGWGetObj::execute(optional_yield y)
     RGWObjManifest m;
     try {
       decode(m, attr_iter->second);
-      if (m.get_tier_type() == "cloud-s3") {
-        /* XXX: Instead send presigned redirect or read-through */
-        op_ret = -ERR_INVALID_OBJECT_STATE;
-        s->err.message = "This object was transitioned to cloud-s3";
-        ldpp_dout(this, 4) << "Cannot get cloud tiered object. Failing with "
-            << op_ret << dendl;
-        goto done_err;
+      op_ret = handle_tier_obj(m, attrs, sync_cloudtiered);
+      if (op_ret < 0) {
+          s->err.message = "This object was transitioned to cloud-s3";
+          ldpp_dout(this, 4) << "Cannot get cloud tiered object. Failing with "
+              << op_ret << dendl;
+          goto done_err;
```

Comment:
can the RGWObjManifest decode and try-catch block move into `handle_tier_obj()`?

Reply:
done.. please review!